Add trailing star to all hwdb patterns

keszybz opened the pull request.

poettering commented on hwdb.d/60-keyboard.hwdb:
```diff
@@ -1820,24 +1821,24 @@ evdev:input:b0003v04FEp000D*
 
 # Lenovo ThinkPad T430s
-evdev:atkbd:dmi:bvn*:bvr*:bd*:svnLENOVO:pn*:pvrThinkPadT430s
+evdev:atkbd:dmi:bvn*:bvr*:bd*:svnLENOVO:pn*:pvrThinkPadT430s*
```
should probably end in `:*`, not just `*`

keszybz replied:
Hmm, am I completely confused, or did this pattern never work? The key that is used for matching has a bunch of trailing stuff (':rvn…:rn…:rvr…:cvn…:ct…:cvr…:'), and the key doesn't match without the trailing asterisk:
```console
$ systemd-hwdb query 'evdev:atkbd:dmi:bvn_:bvr_:bd_:svnLENOVO:pn_:pvrThinkPadT430s' 
KEYBOARD_LED_CAPSLOCK=0
$ systemd-hwdb query 'evdev:atkbd:dmi:bvn_:bvr_:bd_:svnLENOVO:pn_:pvrThinkPadT430s:'
$ systemd-hwdb query 'evdev:atkbd:dmi:bvn_:bvr_:bd_:svnAcer_:pnPeppy:pvr_'
...
XKB_FIXED_MODEL="chromebook"
```
Those were added in c72102edd50, 16bed3afa1b, and bc33509ea0d. @mfvescovi, @paride, @ltworf: did this work back then and stopped working since?

keszybz replied:
With v239-stable (plain old v239 doesn't want to build because of meson changes), I get the same result as with git master. It seems this never worked.

mfvescovi replied:
> Those were added in [c72102e](https://github.com/systemd/systemd/commit/c72102edd503cabfb7abeee009e06b823a9717c4), [16bed3a](https://github.com/systemd/systemd/commit/16bed3afa1b916ace5e927392a1baab9dd9ff963), and [bc33509](https://github.com/systemd/systemd/commit/bc33509ea0d688428f809712f67c9536a34b6356). @mfvescovi, @paride, @ltworf: did this work back then and stopped working since?

Actually, never checked. I was awaiting for the feature to be fully supported in GNOME (my DE of choice) but I guess it was never completed. Thus, no change.

poettering commented on hwdb.d/60-keyboard.hwdb:
```diff
@@ -1820,24 +1821,24 @@ evdev:input:b0003v04FEp000D*
 
 # Lenovo ThinkPad X1 Carbon 3rd Gen
-evdev:atkbd:dmi:bvn*:bvr*:bd*:svnLENOVO:pn*:pvrThinkPadX1Carbon3rd
+evdev:atkbd:dmi:bvn*:bvr*:bd*:svnLENOVO:pn*:pvrThinkPadX1Carbon3rd*
```
these four too. Plenty others that match evdev: and end in a pvr check do place the `:*`at the end, and I think we should do here too

poettering reviewed (changes requested):
looks good. But I think both docs and all matches should use `:*` as suffix instead of just `*`, after all we want to be able to add new fields, but not make the field matches more vague.

poettering commented:
(hmm, isn#t the work "asterisk" rather than "star"? ;-))

keszybz commented:
I thought asterisk is when it's a superscript, and "star" when it's on the level of normal letters. But I might be making this up ;)

poettering commented:
Judging by https://en.wikipedia.org/wiki/Asterisk there's nothing special about superscript or not.

poettering commented:
should the python checkers look for this too?

keszybz commented:
Updated. The suffix is now enforced by the parse_hwdb.py test. It turns out that many changes needed to be made. I split the commit in two: the part where the new patterns are supposed to change matches and a second much bigger part where there should be no immediate difference. I put long descriptions on the commits, please take a look and check my reasoning there.

(Why do I always pick the messiest subject to work on?)

poettering commented:
looks good: the hwdb(7) man page examples should be updated too, though

keszybz commented:
I updated the man page and added another commit to describe wildcard syntax, ptal.
